### ISSUE TITLE
JLV does not seem to work

### ISSUE THREAD

**Issue by rigon** [rdiachenko/jlv#82]
I cannot make JLV to work. The JLV server is listening but does not show any log entry when log4j runs, but JLV server is also up and running:

    $ nc -zv localhost 4445
    Connection to localhost 4445 port [tcp/upnotifyp] succeeded!

Log4j is properly configured, I can grab the output with the netcat (with JLV stopped of course):

    $ nc -l 4445 > temp.log

If I resend the captured log doesn't help, JLV does not show entries:

    $ cat temp.log | nc localhost 4445

I'm using log4j2. My configuration is the same in the example:

	<?xml version="1.0" encoding="UTF-8"?>
	<Configuration>
	  <Appenders>
	    <Socket name="JLV" host="localhost" port="4445">
	      <SerializedLayout />
	    </Socket>
	  </Appenders>
	  <Loggers>
	    <Root level="DEBUG">
	      <AppenderRef ref="JLV" />
	    </Root>
	  </Loggers>
	</Configuration>

And here are my dependencies for log4j2 in `pom.xml`:

	<dependency>
		<groupId>org.apache.logging.log4j</groupId>
		<artifactId>log4j-api</artifactId>
		<version>2.8</version>
	</dependency>
	<dependency>
		<groupId>org.apache.logging.log4j</groupId>
		<artifactId>log4j-core</artifactId>
		<version>2.8</version>
	</dependency>

Can you help me?
Bwt, congrats to your great project!!

**Comment by rdiachenko** [comment]
@rigon , could you please attache the recent plugins logs? They are located in `<path_to_workspace>/.metadata/.plugins/com.rdiachenko.jlv.plugin/logs/jlv-plugin.log`

**Comment by rigon** [comment]
[2017-03-09 11:27:53,286] [INFO] [ResourceManager] [Framework stop]: Font registry disposed
	[2017-03-09 11:27:53,287] [INFO] [ResourceManager] [Framework stop]: Color registry disposed
	[2017-03-09 11:27:53,287] [INFO] [ResourceManager] [Framework stop]: Image registry disposed
	[2017-03-09 11:55:32,850] [INFO] [SourceProvider] [main]: serverStarted source variable initialized to true
	[2017-03-09 11:58:31,280] [INFO] [ResourceManager] [Framework stop]: Font registry disposed
	[2017-03-09 11:58:31,287] [INFO] [ResourceManager] [Framework stop]: Color registry disposed
	[2017-03-09 11:58:31,290] [INFO] [ResourceManager] [Framework stop]: Image registry disposed
	[2017-03-09 11:58:38,377] [INFO] [SourceProvider] [main]: serverStarted source variable initialized to true
	[2017-03-09 12:17:50,604] [INFO] [ResourceManager] [Framework stop]: Font registry disposed
	[2017-03-09 12:17:50,605] [INFO] [ResourceManager] [Framework stop]: Color registry disposed
	[2017-03-09 12:17:50,606] [INFO] [ResourceManager] [Framework stop]: Image registry disposed
	[2017-03-09 12:18:09,272] [INFO] [SourceProvider] [main]: serverStarted source variable initialized to true
	[2017-03-09 12:27:53,240] [INFO] [ResourceManager] [Framework stop]: Font registry disposed
	[2017-03-09 12:27:53,241] [INFO] [ResourceManager] [Framework stop]: Color registry disposed
	[2017-03-09 12:27:53,242] [INFO] [ResourceManager] [Framework stop]: Image registry disposed
	[2017-03-09 12:28:02,016] [INFO] [SourceProvider] [main]: serverStarted source variable initialized to true
	[2017-03-09 12:33:50,977] [INFO] [ResourceManager] [Framework stop]: Font registry disposed
	[2017-03-09 12:33:50,978] [INFO] [ResourceManager] [Framework stop]: Color registry disposed
	[2017-03-09 12:33:50,978] [INFO] [ResourceManager] [Framework stop]: Image registry disposed
	[2017-03-09 12:33:56,852] [INFO] [SourceProvider] [main]: serverStarted source variable initialized to true
	[2017-03-09 13:31:29,133] [INFO] [SourceProvider] [main]: serverStarted source variable initialized to true
	[2017-03-09 14:27:09,741] [INFO] [ResourceManager] [Framework stop]: Font registry disposed
	[2017-03-09 14:27:09,742] [INFO] [ResourceManager] [Framework stop]: Color registry disposed
	[2017-03-09 14:27:09,745] [INFO] [ResourceManager] [Framework stop]: Image registry disposed
	[2017-03-09 14:27:18,310] [INFO] [SourceProvider] [main]: serverStarted source variable initialized to true

**Comment by rdiachenko** [comment]
@rigon, could you please attache the full log? There should be information about server startup and listening for new connections.. I don't see this in the aforementioned piece of log.

Some questions:
1. Which Eclipse version do you use?
2. Do you use slf4j over log4j2? If yes, do you have this dependency in your maven pom:
```
<dependency>
        <groupId>org.apache.logging.log4j</groupId>
        <artifactId>log4j-slf4j-impl</artifactId>
        <version>2.8</version>
</dependency>
```
3. There was a bug #81, did you notice anything similar?

**Comment by rigon** [comment]
That's weird, that is the entire contents of the file... I could not find logs about server statups and shutdowns.

1. My eclipse version is (can be this the problem?):

       Version: Oxygen Release Milestone 5 (4.7.0 M5)
       Build id: 20170202-1353

2. No, I don't. But with that dependency it is the same.

3. I don't notice anything similar, I even changed to dark theme.

When I run my program, I got this error that eventually can help:

	2017-03-09 20:26:50,663 main ERROR Unable to write to stream TCP:localhost:4445 for appender JLV: org.apache.logging.log4j.core.appender.AppenderLoggingException: Error writing to TCP:localhost:4445
	2017-03-09 20:26:50,664 main ERROR An exception occurred processing Appender JLV org.apache.logging.log4j.core.appender.AppenderLoggingException: Error writing to TCP:localhost:4445
		at org.apache.logging.log4j.core.net.TcpSocketManager.write(TcpSocketManager.java:229)
		at org.apache.logging.log4j.core.appender.OutputStreamManager.write(OutputStreamManager.java:201)
		at org.apache.logging.log4j.core.appender.AbstractOutputStreamAppender.writeByteArrayToManager(AbstractOutputStreamAppender.java:186)
		at org.apache.logging.log4j.core.appender.SocketAppender.directEncodeEvent(SocketAppender.java:446)
		at org.apache.logging.log4j.core.appender.AbstractOutputStreamAppender.tryAppend(AbstractOutputStreamAppender.java:170)
		at org.apache.logging.log4j.core.appender.AbstractOutputStreamAppender.append(AbstractOutputStreamAppender.java:161)
		at org.apache.logging.log4j.core.config.AppenderControl.tryCallAppender(AppenderControl.java:156)
		at org.apache.logging.log4j.core.config.AppenderControl.callAppender0(AppenderControl.java:129)
		at org.apache.logging.log4j.core.config.AppenderControl.callAppenderPreventRecursion(AppenderControl.java:120)
		at org.apache.logging.log4j.core.config.AppenderControl.callAppender(AppenderControl.java:84)
		at org.apache.logging.log4j.core.config.LoggerConfig.callAppenders(LoggerConfig.java:448)
		at org.apache.logging.log4j.core.config.LoggerConfig.processLogEvent(LoggerConfig.java:433)
		at org.apache.logging.log4j.core.config.LoggerConfig.log(LoggerConfig.java:417)
		at org.apache.logging.log4j.core.config.LoggerConfig.log(LoggerConfig.java:403)
		at org.apache.logging.log4j.core.config.AwaitCompletionReliabilityStrategy.log(AwaitCompletionReliabilityStrategy.java:63)
		at org.apache.logging.log4j.core.Logger.logMessage(Logger.java:146)
		at org.apache.logging.log4j.spi.AbstractLogger.logMessageSafely(AbstractLogger.java:2091)
		at org.apache.logging.log4j.spi.AbstractLogger.logMessage(AbstractLogger.java:1988)
		at org.apache.logging.log4j.spi.AbstractLogger.logIfEnabled(AbstractLogger.java:1960)
		at org.apache.logging.log4j.spi.AbstractLogger.fatal(AbstractLogger.java:1030)
		at JLV_Test.JLV_Test.App.main(App.java:17)
	Caused by: java.net.SocketException: Connection reset
		at java.net.SocketOutputStream.socketWrite(SocketOutputStream.java:115)
		at java.net.SocketOutputStream.write(SocketOutputStream.java:155)
		at org.apache.logging.log4j.core.net.TcpSocketManager.write(TcpSocketManager.java:219)
		... 20 more

This is my program:

	package JLV_Test.JLV_Test;

	import org.apache.logging.log4j.LogManager;
	import org.apache.logging.log4j.Logger;

	public class App 
	{
		public static void main( String[] args )
		{
			Logger log = LogManager.getLogger(App.class);
			while(true) {
				System.out.println( "Hello World!" );
				log.fatal("FATAL");
				log.error("ERROR");
				log.info("INFO");
				log.debug("DEBUG");
				log.trace("TRACE");
				
				try {
					Thread.sleep(800);
				} catch (InterruptedException e) {
					// TODO Auto-generated catch block
					e.printStackTrace();
				}
			}
		}
	}

And my `pom.xml`:

	<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
		xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
		<modelVersion>4.0.0</modelVersion>

		<groupId>JLV_Test</groupId>
		<artifactId>JLV_Test</artifactId>
		<version>0.0.1-SNAPSHOT</version>
		<packaging>jar</packaging>

		<name>JLV_Test</name>
		<url>http://maven.apache.org</url>

		<properties>
			<project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
		</properties>

		<dependencies>
			<dependency>
				<groupId>junit</groupId>
				<artifactId>junit</artifactId>
				<version>3.8.1</version>
				<scope>test</scope>
			</dependency>
			<dependency>
				<groupId>org.apache.logging.log4j</groupId>
				<artifactId>log4j-api</artifactId>
				<version>2.8.1</version>
			</dependency>
			<dependency>
				<groupId>org.apache.logging.log4j</groupId>
				<artifactId>log4j-core</artifactId>
				<version>2.8.1</version>
			</dependency>
		</dependencies>
	</project>

**Comment by rdiachenko** [comment]
@rigon, checked JLV_Test with that pom you provided on the same Eclipse version as yours, works fine. The aforementioned error says that the client (JLV_Test) can't connect to the server.
```
2017-03-09 20:52:34,003 main ERROR Unable to write to stream TCP:localhost:4445 for appender JLV: org.apache.logging.log4j.core.appender.AppenderLoggingException: Error writing to TCP:localhost:4445
2017-03-09 20:52:34,005 main ERROR An exception occurred processing Appender JLV org.apache.logging.log4j.core.appender.AppenderLoggingException: Error writing to TCP:localhost:4445
	at org.apache.logging.log4j.core.net.TcpSocketManager.write(TcpSocketManager.java:229)
...
```
1. Could you restart JLV server, run JLV_TEST and gather logs from `<path_to_workspace>/.metadata/.plugins/com.rdiachenko.jlv.plugin/logs/`?
2. Could you provide the output of
```
ls -al <path_to_workspace>/.metadata/.plugins/com.rdiachenko.jlv.plugin/logs/
```
3. If you changed JLV preferences could you provide the output of
```
cat <path_to_workspace>/.metadata/.plugins/org.eclipse.core.runtime/.settings/com.rdiachenko.jlv.plugin.prefs 
```

**Comment by rigon** [comment]
I got it working! I installed Eclipse again...

But I found the problem.
Some time ago, I installed version 1.3.3 which doesn't support log4j v2 so could not see the logs.
Later, I noticed the 1.4.0 release I updated Eclipse with the new version.
But the older plugin remained installed and I kept looking at the old JLV view all the time. The new version never actually got activated...

The ID of the plugin 1.3.3 is `com.github.rd.jlv.feature.feature.group` and the 1.4.0 is `com.rdiachenko.jlv.feature.feature.group`. Eclipse did not replaced the old version with the new one.

So, everyone which updates from 1.3.3 to 1.4.0 might experience the same problem.

Now, I got everything working gracefully!
